oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java

```diff
@@ -447,6 +477,117 @@ private void init() {
                 values.add(null);
             }
         }
+
+        public class AlarmStateMachine {
+            @Getter
+            private int silenceCountdown;
+            @Getter
+            private int recoveryObservationCountdown;
+            private final int silencePeriod;
+            private final int recoveryObservationPeriod;
+            private State currentState;
+
+            public AlarmStateMachine(int silencePeriod, int recoveryObservationPeriod) {
+                this.currentState = State.NORMAL;
+                this.silencePeriod = silencePeriod;
+                this.recoveryObservationPeriod = recoveryObservationPeriod;
+                this.silenceCountdown = -1;
+                this.recoveryObservationCountdown = recoveryObservationPeriod;
```

Comment:
If `recoveryObservationPeriod` is not set, according to the following logic, I think the alarm will trigger the recovery immediately if not match?  We need a compatible logical if the user didn't config the recovery settings.